### ISSUE TITLE
Series.values behavior doesn't match pandas

### ISSUE THREAD

**Issue by YarShev** [modin-project/modin#4187]
```python
import modin.pandas as pd
import pandas

ps = pandas.Series(["a", "b", "c", "a"], dtype="category")
print(ps)
0    a
1    b
2    c
3    a
dtype: category
Categories (3, object): ['a', 'b', 'c']

print(ps.values)
['a', 'b', 'c', 'a']
Categories (3, object): ['a', 'b', 'c']

print(ps.values.codes)
[0 1 2 0]

ms = pd.Series(["a", "b", "c", "a"], dtype="category")
print(ms) # output is different against pandas. see more in https://github.com/modin-project/modin/issues/4186
0    a
1    b
2    c
3    a
Name: __reduced__, dtype: category
dtype: category
Categories (3, object): ['a', 'b', 'c']

print(ms.values)
array(['a', 'b', 'c', 'a'], dtype=object)

print(ms.values.codes)
AttributeError: 'numpy.ndarray' object has no attribute 'codes'
```

**Comment by vnlitvinov** [comment]
Related to https://github.com/modin-project/modin/issues/4186

**Comment by jbrockmendel** [comment]
the solution here is something like

```
def values(self) -> ArrayLike:
    arrs = [ser.values for ser in self.pandas_series_in_partitions]
    return pandas.core.dtypes.concat.concat_compat(arrs)
```

is there a code pattern for this?

**Comment by YarShev** [comment]
Closed in #5469.
